core/src/main/java/net/consensys/eventeum/service/DefaultSubscriptionService.java

```diff
@@ -208,26 +208,31 @@ public void unsubscribeFilterSubscription(FilterSubscription filterSubscription)
     }
 
     private ContractEventFilter doRegisterContractEventFilter(ContractEventFilter filter, boolean broadcast) {
-        populateIdIfMissing(filter);
-
-        if (!isFilterRegistered(filter)) {
-            final FilterSubscription sub = registerContractEventFilter(filter, filterSubscriptions);
-
-            if (filter.getStartBlock() == null && sub != null) {
-                filter.setStartBlock(sub.getStartBlock());
-            }
-
-            saveContractEventFilter(filter);
-
-            if (broadcast) {
-                broadcastContractEventFilterAdded(filter);
-            }
-
-            return filter;
-        } else {
-            log.info("Already registered contract event filter with id: " + filter.getId());
-            return getFilterSubscription(filter.getId()).getFilter();
-        }
+        try {
```

Comment:
Just wrapping the current login inside a try-catch block, so it logs the errors when something fails during the eternal retry.